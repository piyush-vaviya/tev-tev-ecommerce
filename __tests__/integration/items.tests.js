const { createTestClient } = require('apollo-server-testing')
const { gql } = require('apollo-server-express')

const { apollo, database } = require('../../app')

const mongo = require('../__mongoserver')

const CREATE_ITEM = gql`
  mutation($input: CreateItemInput!) {
    createItem(input: $input) {
      id
      item_name
      item_description
      item_regular_price
      item_special_price
      item_vendor
      item_country
      item_material
      item_categories
      item_colors
      item_sizes
      sku_prefix
      average_score
      total_reviews
    }
  }
`

const ITEM_CONSTANTS = require('../../constants/items')

describe('Items API', () => {
  const client = createTestClient(apollo)
  beforeAll(async () => {
    const URI = await mongo.connect()

    await database({
      host: URI,
    })
  })

  it('create a new item', async () => {
    const res = await client.mutate({
      query: CREATE_ITEM,
      variables: {
        input: {
          item_name: 'Men Premium T-shirt',
          item_description: 'A cotton tshirt for men',
          item_regular_price: 9.99,
          item_special_price: 9.99,
          sku_prefix: 'mpt',
          item_colors: ['blue', 'royalblue', 'white_n_green', 'forestgreen'],
          item_sizes: ITEM_CONSTANTS.SIZES,
          item_categories: ITEM_CONSTANTS.CATEGORIES,
        },
      },
    })

    expect(res).toMatchSnapshot()
  })
})
