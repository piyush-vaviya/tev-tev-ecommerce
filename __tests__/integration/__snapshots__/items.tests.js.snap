// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items API create a new item 1`] = `
Object {
  "data": Object {
    "createItem": Object {
      "average_score": null,
      "created_at": 1585603874526,
      "id": "5e8265229582021e2a1b36e6",
      "item_categories": Array [
        "Men",
        "Women",
        "Kids",
        "Home",
      ],
      "item_colors": Array [
        "blue",
        "royalblue",
        "white_n_green",
        "forestgreen",
      ],
      "item_country": null,
      "item_description": "A cotton tshirt for men",
      "item_material": null,
      "item_name": "Men Premium T-shirt",
      "item_regular_price": 9.99,
      "item_sizes": Array [
        "Free Size",
        "S",
        "M",
        "L",
        "1XL",
        "2XL",
        "3XL",
        "4XL",
        "5XL",
        "3-6Months",
        "6-9Months",
        "9-12Months",
        "2T",
        "3T",
        "4T",
        "5T",
        "6T",
        "7T",
        "8T",
      ],
      "item_special_price": 9.99,
      "item_vendor": null,
      "sku_prefix": "mpt",
      "total_reviews": null,
      "updated_at": 1585603874526,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
